### PR TITLE
Fix commit.Identity and add test

### LINKED ISSUE #1291: Commit Identity function isn't consisted
The flow

```go
commitID := graveler.CommitExistingMetaRange(metarangeID)
graveler.UpdateBranch(branch, commitID)
```

is broken since UpdateBranch uses RevParse which seems to output different result for commitID.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1291

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]

File: go.sum [hunk 1]
```diff
@@ -197,8 +197,6 @@ github.com/cockroachdb/errors v1.8.2 h1:rnnWK9Nn5kEMOGz9531HuDx/FOleL4NVH20VsDex
 github.com/cockroachdb/errors v1.8.2/go.mod h1:qGwQn6JmZ+oMjuLwjWzUNqblqk0xl4CVV3SQbGwK7Ac=
 github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f h1:o/kfcElHqOiXqcou5a3rIlMc7oJbMQkeLk0VQJ7zgqY=
 github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f/go.mod h1:i/u985jwjWRlyHXQbwatDASoW0RMlZ/3i9yJHE2xLkI=
-github.com/cockroachdb/pebble v0.0.0-20201130172119-f19faf8529d6 h1:nDRfCYBBgSZ59ap7N5jxn0tqJZ1q4Q8Bs13a2ky1q3Y=
```
Comment (arielshaqed) [inline go.sum:200]: Why did this file change?

File: graveler/ref/manager_test.go [hunk 2]
```diff
@@ -610,6 +611,37 @@ func (f *fakeAddressProvider) ContentAddress(_ ident.Identifiable) string {
...
 }
 
+func TestConsistentCommitIdentity(t *testing.T) {
+	hex := ident.NewHexAddressProvider()
+
+	commit := graveler.Commit{
+		Committer:    "some-committer",
+		Message:      "I just committed",
+		MetaRangeID:  "123456789987654321",
+		CreationDate: time.Date(2021, time.January, 24, 15, 10, 11, 0, time.UTC),
```
Comment (arielshaqed) [inline graveler/ref/manager_test.go:621]: Can we also have some microseconds, even nanoseconds, on this time?

File: graveler/ref/manager_test.go [hunk 3]
```diff
@@ -610,6 +611,37 @@ func (f *fakeAddressProvider) ContentAddress(_ ident.Identifiable) string {
...
+		},
+		Metadata: map[string]string{
+			"sdkafjnb":       "1234",
+			"sdkjvcnbkjndsc": "asnjkdl",
+		},
+	}
+
+	// Should NOT be changed (unless you really know what you're doing):
+	// If this is failing and you're tempted to change this value,
+	// then you probably introducing a braking change to commit's identity.
```
Comment (arielshaqed) [inline graveler/ref/manager_test.go:634]:
```suggestion
	// then you probably introducing a breaking change to commit's identity.
```
# :stop_sign:

File: graveler/ref/manager_test.go [hunk 4]
```diff
@@ -610,6 +611,37 @@ func (f *fakeAddressProvider) ContentAddress(_ ident.Identifiable) string {
...
+	// All previous references to commits (if not migrated) will be potently lost.
```
Comment (arielshaqed) [inline graveler/ref/manager_test.go:635]:
```suggestion
	// All previous references to commits (if not migrated) will potentially be lost.
```
or just
```suggestion
	// All previous references to commits (if not migrated) may be lost.
```

File: graveler/ref/manager_test.go [hunk 5]
```diff
@@ -610,6 +611,37 @@ func (f *fakeAddressProvider) ContentAddress(_ ident.Identifiable) string {
...
+	const expected = "2a7c0499ffc766c1e47aff0dd0535c8e74a9c89bfc6035339e4700737346f4e3"
+
+	const iterations = 10000
```
Comment (arielshaqed) [inline graveler/ref/manager_test.go:638]: Best add a comment, otherwise nobody will ever believe you why you think a digest computation can change :-)

**Review by arielshaqed — APPROVED** [review]
Yay!

File: go.sum (hunk 1 above)
Comment (itaiad200) [reply go.sum:200]: `go mod tidy`
I was surprised too.

File: graveler/ref/manager_test.go (hunk 2 above)
Comment (itaiad200) [reply graveler/ref/manager_test.go:621]: Yes you can.

File: graveler/ref/manager_test.go (hunk 3 above)
Comment (itaiad200) [reply graveler/ref/manager_test.go:634]: Better than barking change..

**Pull request merged**